lms/djangoapps/verify_student/ssencrypt.py

```diff
@@ -51,68 +54,75 @@ def aes_encrypt(data, key):
     """
     cipher = aes_cipher_from_key(key)
     padded_data = pad(data)
-    return cipher.encrypt(padded_data)
+    encryptor = cipher.encryptor()
+    return encryptor.update(padded_data) + encryptor.finalize()
 
 
 def aes_decrypt(encrypted_data, key):
     """
     Decrypt `encrypted_data` using `key`
     """
     cipher = aes_cipher_from_key(key)
-    padded_data = cipher.decrypt(encrypted_data)
+    decryptor = cipher.decryptor()
+    padded_data = decryptor.update(encrypted_data) + decryptor.finalize()
     return unpad(padded_data)
 
 
 def aes_cipher_from_key(key):
     """
-    Given an AES key, return a Cipher object that has `encrypt()` and
-    `decrypt()` methods. It will create the cipher to use CBC mode, and create
+    Given an AES key, return a Cipher object that has `encryptor()` and
+    `decryptor()` methods. It will create the cipher to use CBC mode, and create
     the initialization vector as Software Secure expects it.
     """
-    return AES.new(key, AES.MODE_CBC, generate_aes_iv(key))
+    return Cipher(AES(key), CBC(generate_aes_iv(key)), backend=default_backend())
 
 
 def generate_aes_iv(key):
     """
     Return the initialization vector Software Secure expects for a given AES
     key (they hash it a couple of times and take a substring).
     """
-    return md5(key + md5(key).hexdigest()).hexdigest()[:AES.block_size]
+    return md5(key + md5(key).hexdigest()).hexdigest()[:AES_BLOCK_SIZE_BYTES]
 
 
 def random_aes_key():
-    return Random.new().read(32)
+    return os.urandom(32)
 
 
 def pad(data):
     """ Pad the given `data` such that it fits into the proper AES block size """
-    bytes_to_pad = AES.block_size - len(data) % AES.block_size
-    return data + (bytes_to_pad * chr(bytes_to_pad))
+    padder = PKCS7(AES.block_size).padder()
+    return padder.update(data) + padder.finalize()
 
 
 def unpad(padded_data):
     """  remove all padding from `padded_data` """
-    num_padded_bytes = ord(padded_data[-1])
-    return padded_data[:-num_padded_bytes]
+    unpadder = PKCS7(AES.block_size).unpadder()
+    return unpadder.update(padded_data) + unpadder.finalize()
 
 
 def rsa_encrypt(data, rsa_pub_key_str):
     """
-    `rsa_pub_key` is a string with the public key
+    `rsa_pub_key` is a byte sequence with the public key
```

Comment:
Docstring name doesn't match param name.

Reply:
I'll update this to match the new param name.